scipy/stats/_distn_infrastructure.py

```diff
@@ -2876,6 +2879,129 @@ def entropy(pk, qk=None, base=None, axis=0):
     return S
 
 
+def differential_entropy(
+    values: np.typing.ArrayLike,
+    *,
+    window_length: Optional[int] = None,
+    base: Optional[float] = None,
+    axis: int = 0,
+) -> np.ndarray:
```

Comment:
Mypy complains about this:

```none
error: Incompatible return value type (got "Union[number[Any], ndarray]", expected "ndarray")  [return-value]
```

https://github.com/scipy/scipy/pull/13448#discussion_r572137286 suggests to use `Any` here. But we can also be compatible with NumPy's type annotation for `np.mean` which is `Union[number[Any], ndarray]`. IMO, the latter annotation is more useful as it gives more information.

If you agree, can you please change this to `Any` or `Union[np.number, np.ndarray]`? If you do this, also please remove `_distn_infrastructure.py` from `mypy.ini` file so that it gets checked for typing errors.

Reply:
I have done as you suggested.